pkg/engine/inspector.go

```diff
@@ -137,14 +137,25 @@ func NewInspector(
 		default:
 			var opaQuery rego.PreparedEvalQuery
 			store := inmem.NewFromReader(bytes.NewBufferString(metadata.InputData))
-			opaQuery, err = rego.New(
-				rego.Query(regoQuery),
-				rego.Module("Common", commonGeneralQuery),
-				rego.Module("Generic", platformGeneralQuery),
-				rego.Module(metadata.Query, metadata.Content),
-				rego.Store(store),
-				rego.UnsafeBuiltins(unsafeRegoFunctions),
-			).PrepareForEval(ctx)
+
+			if commonGeneralQuery != "" {
+				opaQuery, err = rego.New(
+					rego.Query(regoQuery),
+					rego.Module("Common", commonGeneralQuery),
+					rego.Module("Generic", platformGeneralQuery),
+					rego.Module(metadata.Query, metadata.Content),
+					rego.Store(store),
+					rego.UnsafeBuiltins(unsafeRegoFunctions),
+				).PrepareForEval(ctx)
+			} else {
```

Comment:
should we check if platform-specific queries are not available too with `platformGeneralQuery != ""`